mcs/class/System/System.Net.Sockets/Socket.cs

```diff
@@ -1172,6 +1172,10 @@ bool GetCheckedIPs (SocketAsyncEventArgs e, out IPAddress [] addresses)
 			DnsEndPoint dep = e.RemoteEndPoint as DnsEndPoint;
 			if (dep != null) {
 				addresses = Dns.GetHostAddresses (dep.Host);
+
+				if (dep.AddressFamily == AddressFamily.Unspecified)
```

Comment:
This could be done even before `Dns.GetHostAddresses`

Reply:
@marek-safar This can't be done before Dns.GetHostAddresses, as that populates the addresses field that we want to return. by returning out after the call to Dns.GetHostAddresses, we return the full list of addresses instead of filtering out unmatching address below.